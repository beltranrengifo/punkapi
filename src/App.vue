<template>
  <v-app>
    <pa-header />
    <pa-drawer />
    <v-main>
      <!-- Provides the application the proper gutter -->
      <v-container fluid></v-container>
    </v-main>

    <v-footer app>
      <pa-theme-switcher />
    </v-footer>
  </v-app>
</template>

<script>
import { mapActions } from 'vuex'
import PaHeader from '@/components/PaHeader'
import PaDrawer from '@/components/PaDrawer'
import PaThemeSwitcher from '@/components/PaThemeSwitcher'

export default {
  name: 'App',

  components: {
    PaHeader,
    PaDrawer,
    PaThemeSwitcher,
  },

  async created() {
    this.setTitle()
    await this.fetchBeers()
  },

  methods: {
    setTitle() {
      document.title = this.$store.state.common.DEFAULT_TITLE
    },

    ...mapActions({
      fetchBeers: 'punkapi/fetchBeers',
    }),
  },
}
</script>
