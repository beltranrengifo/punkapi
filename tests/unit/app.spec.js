import Vue from 'vue'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

import { shallowMount, createLocalVue } from '@vue/test-utils'

import App from '@/App'

const localVue = createLocalVue()
Vue.use(Vuetify)
localVue.use(Vuex)

describe('App', () => {
  let actions
  let store
  let wrapper

  beforeEach(() => {
    actions = { 'punkapi/fetchBeers': jest.fn() }
    store = new Vuex.Store({ actions })
    wrapper = shallowMount(App, { localVue, store })
  })

  it('App works and is mounting Vuetify components', () => {
    expect(wrapper.findComponent({ name: 'v-app' })).toBeTruthy()
  })

  it('App should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Dispatches fetchBeers when instance is created', () => {
    expect(actions['punkapi/fetchBeers']).toHaveBeenCalled()
  })
})
