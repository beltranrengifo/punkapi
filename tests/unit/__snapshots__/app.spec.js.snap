// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match the snapshot 1`] = `
<v-app-stub
  id="app"
>
  <pa-header-stub />
   
  <pa-drawer-stub />
   
  <v-main-stub
    tag="main"
  >
    <v-container-stub
      fluid="true"
      tag="div"
    />
  </v-main-stub>
   
  <v-footer-stub
    app="true"
    height="auto"
    tag="footer"
  >
    <v-btn-stub
      activeclass=""
      icon="true"
      tag="button"
      type="button"
    >
      <v-icon-stub>
        mdi-theme-light-dark
      </v-icon-stub>
    </v-btn-stub>
  </v-footer-stub>
</v-app-stub>
`;
